### PR TITLE
Implementation of request split for AppNexus adapter

### PR DESCRIPTION
**Pull request by VeronikaSolovei9**

### CODE REVIEW

File: adapters/appnexus/appnexus.go [hunk 1]
```diff
@@ -345,21 +347,40 @@ func (a *AppNexusAdapter) MakeRequests(request *openrtb.BidRequest) ([]*adapters
...
-	}}, errs
+		reqJSON, err := json.Marshal(request)
+		if err != nil {
+			errs = append(errs, err)
+			return nil, errs
+		}
+
+		headers := http.Header{}
+		headers.Add("Content-Type", "application/json;charset=utf-8")
+		headers.Add("Accept", "application/json")
```
Comment (jmaynardxandr) [inline adapters/appnexus/appnexus.go]: You can move headers outside of the for loop so they only get setup once.

File: adapters/appnexus/appnexus.go [hunk 2]
```diff
@@ -345,21 +347,40 @@ func (a *AppNexusAdapter) MakeRequests(request *openrtb.BidRequest) ([]*adapters
 		}
 	}
 
-	reqJSON, err := json.Marshal(request)
-	if err != nil {
-		errs = append(errs, err)
-		return nil, errs
-	}
+	imps := request.Imp
+	resArr := make([]*adapters.RequestData, 0)
```
Comment (jmaynardxandr) [inline adapters/appnexus/appnexus.go]: Could create resArr with an initial capacity of `(len(imps)+maxImpsPerReq-1)/maxImpsPerReq` so it allocates memory for all needed requests up front.

**Review by jmaynardxandr — APPROVED** [review]

File: adapters/appnexus/appnexus.go (hunk 1 above)
Comment (VeronikaSolovei9) [reply adapters/appnexus/appnexus.go]: Done!

File: adapters/appnexus/appnexus.go (hunk 2 above)
Comment (VeronikaSolovei9) [reply adapters/appnexus/appnexus.go]: Done!

**Pull request merged**